backtracepython/report_queue.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import threading
+
+if sys.version_info.major >= 3:
+    import queue
+else:
+    import Queue as queue
+
+
+class ReportQueue:
+    def __init__(self, request_handler, source_code_handler):
+        self.request_handler = request_handler
+        self.source_code_handler = source_code_handler
+
+        # report submission tasks queue
+        self.report_queue = queue.Queue()
+
+        # Create and start a single worker thread
+        self.worker_thread = threading.Thread(target=self._worker)
+        self.worker_thread.daemon = True
+        self.active = True
+        self.worker_thread.start()
+
+    def _worker(self):
+        while True:
+            report_data = self.report_queue.get()
+            if report_data is None or self.active == False:
+                self.report_queue.task_done()
+                break
+            report, attachments = report_data
+            self.process(report, attachments)
+            self.report_queue.task_done()
+
+    def add(self, report, attachments):
+        self.report_queue.put((report, attachments))
```

Comment:
you'll want to use `put_nowait` here probably

Reply:
that's true. Changing it. Thanks